Feature/#12: 메인 화면 채널 및 장소 라벨 컴포넌트 및 스토리북 구현

minjeongss opened the pull request:
## #️⃣ 연관된 이슈
#12 

## 📝 작업 내용
### 메인 화면
메인 화면에서 사용되는 컴포넌트 구현 및 스토리북 구현을 완료했습니다.
구현 완료한 컴포넌트는 다음과 같습니다.

  ✅ 채널 컴포넌트(게임1, 게임2)
  ✅ 장소 컴포넌트(올린 장소, 저장한 장소)
### 채널 화면
메인 화면에서 사용되는 컴포넌트 구현 및 스토리북 구현을 완료했습니다.
구현 완료한 컴포넌트는 다음과 같습니다.

 ✅ 방 컴포넌트
 
이때, 채널에 대한 라우터 설정에 대한 논의를 하지 않아 우선적으로 home의 page.tsx에서 사용했습니다.

## 💬 리뷰 요구사항
- 각 컴포넌트의 prop 수가 적절한지 확인해주세요!
  -  공통적으로 사용되는 컴포넌트의 경우, 변경이 예상되는 부분을 prop에 등록했습니다
  - 하나의 화면에서만 사용되는 컴포넌트의 경우, 최대한 prop을 줄였습니다
=> 이 부분에 대한 코드는 하단에 언급해두었습니다! 🫡

minjeongss committed:
docs: main 및 signup에서 활용되는 icon 추가

minjeongss committed:
feat: home 화면의 Channel 컴포넌트 구현 및 스토리북 구현

minjeongss committed:
feat: home 화면의 장소 컴포넌트 구현 및 스토리북 구현

minjeongss committed:
feat: Channel props 감소

minjeongss committed:
style: Home 내부 컴포넌트 Button을 추가하여 이름 명시적으로 변경

minjeongss committed:
feat: 채널 화면에서 방 컴포넌트 구현 및 스토리북 구현

minjeongss commented on components/Common/RoomButton/RoomButton.tsx:
```diff
@@ -0,0 +1,41 @@
+  totalCount: number;
+}
+
+const RoomButton = ({
+  title,
+  userName,
+  currCount,
+  totalCount,
+}: RoomButtonProps) => {
+  return (
+    <RoomButtonWrapper>
+      <RoomButtonLayout>
+        <div>
+          <RoomTitle>{title}</RoomTitle>
+          <RoomPerson>
+            <PersonIcon />
+            <div>{userName}</div>
+          </RoomPerson>
+        </div>
+        <RoomCount>
+          ({currCount}/{totalCount})
+        </RoomCount>
+      </RoomButtonLayout>
+    </RoomButtonWrapper>
+  );
+};
```
여러 채널에서 사용되는 공통 컴포넌트는 title, userName, currCount, totalCount를 prop으로 등록해서 확장성을 높였습니다.

bluedog129 replied:
<div>{userName}</div>
이 부분이 나중에 여러명의 유저 이름이 들어가야 할 때, 하나의 문자열로 받아야 하는 형식인거 같은데 의도 하고 넣으신 걸까요?

minjeongss replied:
대표 유저만 보이고, 나머지는 `대표유저 외 n명`으로 수정 완료했습니다!

minjeongss commented on components/Layout/Home/ChannelButton/ChannelButton.tsx:
```diff
@@ -0,0 +1,18 @@
+  height?: 'long' | 'short';
+}
+
+const Channel = ({ title, height = 'long' }: ChannelProps) => {
+  return (
+    <ChannelWrapper $height={height}>
+      <ChannelTitle>{title}</ChannelTitle>
+      {height === 'long' ? <EntranceIcon /> : <LockIcon />}
+    </ChannelWrapper>
+  );
+};
```
Home 화면에서만 사용되는 컴포넌트는 최대한 prop을 줄이는 것을 목표로 height에 따라 background도 코드 내에서 자동적으로 처리될 수 있도록 구현했습니다.

minjeongss committed:
feat: RoomButton의 prop 이름 특성에 맞게 변경

- userName > hostUser
- currCount > groupMemberCount
- totalCount > maxMemberCount

aaahyesu commented on components/Common/RoomButton/RoomButton.tsx:
```diff
@@ -0,0 +1,43 @@
+            <div>
+              {hostUser}외 {groupMemberCount - 1}명
+            </div>
+          </RoomPerson>
```
`<div>` 태그 대신 `<RoomWrapper> </RoomWrapper>`로 묶어주시는것이 더 자연스러울 것 같습니다-!

minjeongss replied:
div 대신 `RoomTopContainer`로 변경했습니다! 🫡
```tsx
        <RoomTopContainer>
          <RoomTitle>{title}</RoomTitle>
          <RoomPerson>
            <PersonIcon />
            <PersonBox>
              {hostUser}외 {groupMemberCount - 1}명
            </PersonBox>
          </RoomPerson>
```

aaahyesu commented on components/Layout/Home/ChannelButton/ChannelButton.tsx:
```diff
@@ -0,0 +1,17 @@
+import { ChannelWrapper, ChannelTitle } from './ChannelButton.styles';
+
+interface ChannelProps {
+  title: '게임1' | '게임2';
```
일반 게임 | 랭킹 게임으로 명칭을 바꾸면 좋을 것 같아요 !

minjeongss replied:
그룹 게임 | 랭킹 게임으로 변경 완료했습니다! ☺️👍

aaahyesu reviewed (approved):
확인했습니당-! 메인에 쓰일 요소들 만들어주셔서 감사해요 잘쓸게요오!👍

minjeongss committed:
fix: StyledComponent 적용 & Channel 종류 부여

- Channel 종류: 그룹 게임 , 랭킹 게임
